power_grid_model_c/power_grid_model_cpp/include/power_grid_model_cpp/serialization.hpp

```diff
@@ -0,0 +1,84 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+#ifndef POWER_GRID_MODEL_CPP_SERIALIZATION_HPP
+#define POWER_GRID_MODEL_CPP_SERIALIZATION_HPP
+
+#include "power_grid_model_c/serialization.h"
+
+#include "basics.hpp"
+#include "dataset.hpp"
+#include "handle.hpp"
+
+namespace power_grid_model_cpp {
+class Deserializer {
+  public:
+    Deserializer(std::vector<std::byte> const& data, Idx serialization_format)
+        : deserializer_{PGM_create_deserializer_from_binary_buffer(
+              handle_.get(), reinterpret_cast<char const*>(data.data()), data.size(), serialization_format)} {}
+    Deserializer(std::string const& data_string, Idx serialization_format)
+        : deserializer_{PGM_create_deserializer_from_null_terminated_string(handle_.get(), data_string.c_str(),
+                                                                            serialization_format)} {}
+
+    RawDeserializer* get() const { return deserializer_.get(); }
+
+    static DatasetWritable get_dataset(Deserializer const& deserializer) {
+        RawWritableDataset* dataset = PGM_deserializer_get_dataset(deserializer.handle_.get(), deserializer.get());
+        deserializer.handle_.check_error();
+        return DatasetWritable(dataset);
+    }
+    DatasetWritable get_dataset() const { return get_dataset(*this); }
+
+    static void parse_to_buffer(Deserializer& deserializer) {
+        PGM_deserializer_parse_to_buffer(deserializer.handle_.get(), deserializer.get());
+        deserializer.handle_.check_error();
+    }
+    void parse_to_buffer() { return parse_to_buffer(*this); }
+
+  private:
+    friend class DatasetWritable;
```

Comment:
if possible, please remove `friend`

Reply:
Removed. As the constructor of `DatasetWritable` is now public.